### PR TITLE
Add Torq Connector

### PR DESCRIPTION
**Pull request by natexcvi**
## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### CODE REVIEW

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+      configurationUtilities: {
+        ...actionsConfigMock.create(),
+        ensureUriAllowed: (_) => {
+          throw new Error(`target url is not present in allowedHosts`);
+        },
+      },
+    });
+
+    // any for testing
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (Guy2co) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: why disable the linter?

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    expect(actionType.renderParameterTemplates).toBeTruthy();
+    const paramsWithTemplates = {
+      body: '{"x": "{{rogue}}"}',
+    };
+    const variables = {
+      rogue,
+    };
+    const params = actionType.renderParameterTemplates!(paramsWithTemplates, variables);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (Guy2co) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:248]: why disable the linter

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    let paramsObject: any;
+    try {
+      paramsObject = JSON.parse(`${params.body}`);
+    } catch (err) {
+      expect(err).toBe(null); // kinda weird, but test should fail if it can't parse
```
Comment (Guy2co) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: remove comment

File: x-pack/plugins/actions/server/builtin_action_types/torq.ts [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+  logger.debug(`torq action result: ${JSON.stringify(result)}`);
+
+  if (isOk(result)) {
+    const {
+      value: { status, statusText },
+    } = result;
+    logger.debug(`response from Torq action "${actionId}": [HTTP ${status}] ${statusText}`);
+
+    return successResult(actionId, data);
+  } else {
```
Comment (Guy2co) [inline x-pack/plugins/actions/server/builtin_action_types/torq.ts]: you can remove the else as you have a clause of
```
if (something){
return x
}
// this code won't execute if something is true
```

File: x-pack/plugins/actions/server/builtin_action_types/torq.ts [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+      logger.error(`error on ${actionId} Torq event: ${message}`);
+      return errorResultRequestFailed(actionId, message);
+    } else if (error.isAxiosError) {
+      const message = `${error.message}`;
+      logger.error(`error on ${actionId} Torq event: ${message}`);
+      return errorResultRequestFailed(actionId, message);
+    }
+
+    logger.error(`error on ${actionId} Torq action: unexpected error`);
+    return errorResultUnexpectedError(actionId);
```
Comment (Guy2co) [inline x-pack/plugins/actions/server/builtin_action_types/torq.ts]: we can extract this to a function `getErrorType`

File: x-pack/plugins/actions/server/plugin.ts [hunk 6]
```diff
@@ -328,6 +329,15 @@ export class ActionsPlugin implements Plugin<PluginSetupContract, PluginStartCon
       actionsConfigUtils,
     });
 
+    // subActionFramework.registerConnector({
```
Comment (Guy2co) [inline x-pack/plugins/actions/server/plugin.ts]: remove

File: x-pack/plugins/actions/server/sub_action_framework/torq.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      schema: schema.object({ id: schema.string() }),
+    });
+  }
+
+  protected getResponseErrorMessage(error: AxiosError<ErrorSchema>) {
+    return `Message: ${error.response?.data.errorMessage}. Code: ${error.response?.data.errorCode}`;
+  }
+
+  public async sendToTorqAction({ id  }: { id: string; }) {
+    // const res = await this.request({
```
Comment (Guy2co) [inline x-pack/plugins/actions/server/sub_action_framework/torq.ts]: remove

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_params.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+      onBlur={() => {
+        if (!body) {
+          editAction('body', '', index);
+        }
+      }}
+    />
+  );
+};
+
+// eslint-disable-next-line import/no-default-export
```
Comment (Guy2co) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_params.tsx:42]: why disable eslint?

File: x-pack/plugins/triggers_actions_ui/public/application/lib/get_defaults_for_action_params.ts [hunk 9]
```diff
@@ -12,13 +12,15 @@ import { AlertProvidedActionVariables } from './action_variables';
...
-  actionGroupId: string
+  actionGroupId: string,
+  isRecoveryActionGroup: boolean,
 ) => DefaultActionParams;
 export const getDefaultsForActionParams = (
   actionTypeId: string,
   actionGroupId: string,
   isRecoveryActionGroup: boolean
 ): DefaultActionParams => {
+  console.log('get defaults called');
```
Comment (Guy2co) [inline x-pack/plugins/triggers_actions_ui/public/application/lib/get_defaults_for_action_params.ts]: remove

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/translations.ts [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  'xpack.triggersActionsUI.components.builtinActionTypes.torqAction.selectMessageText',
+  {
+    defaultMessage: 'Trigger a Torq workflow.',
+  }
+);
+
+export const TORQ_ACTION_TYPE_TITLE = i18n.translate(
+  'xpack.triggersActionsUI.components.builtinActionTypes.torqAction.actionTypeTitle',
+  {
+    defaultMessage: 'Alert data', // TODO: other message?
```
Comment (Guy2co) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/translations.ts]: remove todo

**Review by Guy2co — CHANGES_REQUESTED** [review]

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts (hunk 1 above)
Comment (natexcvi) [reply x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: The tests use `Record<string, any>` which the linter wouldn't like. It's a pattern that appears in other tests there too.

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts (hunk 2 above)
Comment (natexcvi) [reply x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:248]: Same as above

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_params.tsx (hunk 8 above)
Comment (natexcvi) [reply x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_params.tsx:42]: This is how other connectors export the params component.

File: x-pack/package.json [hunk 11]
```diff
@@ -23,4 +23,4 @@
   "engines": {
     "yarn": "^1.21.1"
   }
-}
\ No newline at end of file
+}
```
Comment (UrielShimony) [inline x-pack/package.json]: remove redundant space

File: x-pack/plugins/actions/server/builtin_action_types/torq.ts [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+import { ActionsConfigurationUtilities } from '../actions_config';
+import { request } from './lib/axios_utils';
+import { renderMustacheString } from '../lib/mustache_renderer';
+import {
+  AlertingConnectorFeatureId,
+  UptimeConnectorFeatureId,
+  SecurityConnectorFeatureId,
+} from '../../common';
+
+export type TorqActionType = ActionType<
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.ts:28]: I would suggest export all types to .d.ts file

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+jest.mock('./lib/axios_utils', () => {
+  const originalUtils = jest.requireActual('./lib/axios_utils');
+  return {
+    ...originalUtils,
+    request: jest.fn(),
+    patch: jest.fn(),
+  };
+});
+
+axios.create = jest.fn(() => axios);
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: it seems to appear twice

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 14]
```diff
@@ -0,0 +1,254 @@
...
+const requestMock = utils.request as jest.Mock;
+
+axios.create = jest.fn(() => axios);
+
+const ACTION_TYPE_ID = '.torq';
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: I would suggest to put consts  on the top of the file

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 15]
```diff
@@ -0,0 +1,254 @@
...
+  test('exposes the action as `torq` on its Id and Name', () => {
+    expect(actionType.id).toEqual('.torq');
+    expect(actionType.name).toEqual('Torq');
+  });
+});
+
+describe('secrets validation', () => {
+  test('succeeds when secrets is valid', () => {
+    const secrets: Record<string, string> = {
+      token: 'jfi2fji3ofeaiw34if',
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:67]: `const validSecret = 'jfi2fji3ofeaiw34if'`

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 16]
```diff
@@ -0,0 +1,254 @@
...
+    };
+    expect(validateSecrets(actionType, secrets)).toEqual(secrets);
+  });
+
+  test('fails when secret token is not provided', () => {
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:72]: is there any other validation on token except requiered?

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 17]
```diff
@@ -0,0 +1,254 @@
...
+    const config: Record<string, string | boolean> = {
+      webhookIntegrationUrl: 'https://hooks.torq.io/v1/test',
+    };
+    expect(validateConfig(actionType, config)).toEqual({
+      ...defaultValues,
+      ...config,
+    });
+  });
+
+  ['iamnotavalidurl', 'example.com/do-something'].forEach((url) => {
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: i would create an object looks like {url:string,error:string} and iterate it to reduce the number of tests like table tests in golang

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 18]
```diff
@@ -0,0 +1,254 @@
...
+        ...actionsConfigMock.create(),
+        ensureUriAllowed: (_) => {
+          throw new Error(`target url is not present in allowedHosts`);
+        },
+      },
+    });
+
+    // any for testing
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const config: Record<string, any> = {
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: maybe    const config: Record<string, string> ?

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 19]
```diff
@@ -0,0 +1,254 @@
...
+    expect(() => {
+      validateConfig(actionType, config);
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"error validating action type config: error configuring send to Torq action: target url is not present in allowedHosts"`
+    );
+  });
+});
+
+describe('params validation', () => {
+  test('params validation passes when a valid body is provided', () => {
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:146]: is there any validation? can you add a failed test? (maybe empty body)

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 20]
```diff
@@ -0,0 +1,254 @@
...
+    const params: Record<string, string> = {
+      body: '{"message": "Hello"}',
+    };
+    expect(validateParams(actionType, params)).toEqual({
+      ...params,
+    });
+  });
+});
+
+describe('execute()', () => {
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts]: maybe a more informative name will be more elegant here

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts [hunk 21]
```diff
@@ -0,0 +1,254 @@
...
+      actionId: 'some-id',
+      services,
+      config,
+      secrets: { token: '1234' },
+      params: { body: 'some data' },
+    });
+
+    delete requestMock.mock.calls[0][0].configurationUtilities;
+    expect(requestMock.mock.calls[0][0]).toMatchInlineSnapshot(`
+      Object {
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:190]: maybe export it to another file

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts (hunk 2 above)
Comment (UrielShimony) [reply x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:248]: params must have a type...

File: x-pack/plugins/actions/server/builtin_action_types/torq.ts [hunk 22]
```diff
@@ -0,0 +1,316 @@
...
+        'X-Torq-Token': token || '',
+      },
+      params: {},
+      data,
+      configurationUtilities,
+      logger,
+    })
+  );
+
+  logger.debug(`torq action result: ${inspect(result)}`);
```
Comment (UrielShimony) [inline x-pack/plugins/actions/server/builtin_action_types/torq.ts:176]: inspect output is string?

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.test.tsx [hunk 23]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TypeRegistry } from '../../../type_registry';
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.test.tsx:8]: doesnt they have a key phrase for the father directory?

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+
+export function getActionType(): ActionTypeModel<TorqConfig, TorqSecrets, TorqActionParams> {
+  return {
+    id: '.torq',
+    iconClass: lazy(() => import('./logo')),
+    selectMessage: i18n.TORQ_SELECT_MESSAGE,
+    actionTypeTitle: i18n.TORQ_ACTION_TYPE_TITLE,
+    validateParams: async (
+      actionParams: TorqActionParams
+    ): Promise<GenericValidationResult<TorqActionParams>> => {
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx]: you can create the function above and not in the return value

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+      const translations = await import('./translations');
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx]: are you sure it is ok to use awaited import & lazy import? how did they do it in another files?

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+      const errors = {
+        body: [] as string[],
+      };
+      const validationResult = { errors };
+      validationResult.errors = errors;
+      if (!actionParams.body?.length) {
+        errors.body.push(translations.BODY_REQUIRED);
+      }
+      try {
+        JSON.parse(replaceReferencesWithNumbers(actionParams.body || ""));
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx]: same as here, I would suggest to make the call not in the return...

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx [hunk 27]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx:1]: why this file is not .ts?

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.test.tsx [hunk 28]
```diff
@@ -0,0 +1,75 @@
+/*
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.test.tsx:1]: why the file is .tsx

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.test.tsx [hunk 29]
```diff
@@ -0,0 +1,178 @@
...
+      jest.clearAllMocks();
+    });
+
+    it('connector validation succeeds when connector config is valid', async () => {
+      const { getByTestId } = render(
+        <ConnectorFormTestProvider connector={actionConnector} onSubmit={onSubmit}>
+          <TorqActionConnectorFields
+            readOnly={false}
+            isEdit={false}
+            registerPreSubmitValidator={() => {}}
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.test.tsx]: I think it is more elegant to declare const emptyFunction = ()=>{}  in the top of the file and use it or use any mocks they are using

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts (hunk 2 above)
Comment (UrielShimony) [reply x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:248]: or maybe use Record

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.tsx [hunk 30]
```diff
@@ -0,0 +1,100 @@
...
+const torqWebhookEndpoint =
+  (message: string) =>
+  (...args: Parameters<ValidationFunc>): ReturnType<ValidationFunc<any, ERROR_CODE>> => {
+    const [{ value }] = args;
+    const error: ValidationError<ERROR_CODE> = {
+      code: 'ERR_FIELD_FORMAT',
+      formatType: 'URL',
+      message,
+    };
+    // TODO: Fix casting
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.tsx]: ?

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.tsx [hunk 31]
```diff
@@ -0,0 +1,100 @@
...
+      <EuiCallOut size="s" iconType="iInCircle" data-test-subj={dataTestSubj} title={title} />
+      <EuiSpacer size="m" />
+    </>
+  );
+};
+
+const torqWebhookEndpoint =
+  (message: string) =>
+  (...args: Parameters<ValidationFunc>): ReturnType<ValidationFunc<any, ERROR_CODE>> => {
+    const [{ value }] = args;
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.tsx]: you can declare the value type as string here

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.tsx [hunk 32]
```diff
@@ -0,0 +1,100 @@
...
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSpacer size="m" />
+      <EuiSpacer size="m" />
+    </>
+  );
+};
+
+// eslint-disable-next-line import/no-default-export
+export { TorqActionConnectorFields as default };
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq_connectors.tsx:99]: export  default TorqActionConnectorFields

File: x-pack/plugins/triggers_actions_ui/public/application/lib/get_defaults_for_action_params.ts [hunk 33]
```diff
@@ -38,5 +39,59 @@ export const getDefaultsForActionParams = (
         spaceId: `{{${AlertProvidedActionVariables.ruleSpaceId}}}`,
       };
       return xmattersDefaults;
+    case '.torq':
+      const torqDefaults = {
+        body: JSON.stringify(
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/lib/get_defaults_for_action_params.ts]: why we are the only one who use  JSON.stringify?

Comment (UrielShimony) [reply x-pack/plugins/triggers_actions_ui/public/application/lib/get_defaults_for_action_params.ts]: you can export this thing to another file and return it here

File: x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx [hunk 34]
```diff
@@ -361,7 +360,7 @@ export const ActionForm = ({
               messageVariables={messageVariables}
               actionGroups={actionGroups}
               defaultActionMessage={defaultActionMessage}
-              defaultParams={getDefaultActionParams?.(actionItem.actionTypeId, actionItem.group)}
+              defaultParams={getDefaultActionParams?.(actionItem.actionTypeId, actionItem.group, false)}
```
Comment (UrielShimony) [inline x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx:363]: why did we do this change?

**Review by UrielShimony — CHANGES_REQUESTED** [review]

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts (hunk 16 above)
Comment (natexcvi) [inline x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:71]: Nope, that's the only static validation. Anything else you believe we should check here?

File: x-pack/plugins/actions/server/builtin_action_types/torq.test.ts (hunk 19 above)
Comment (natexcvi) [reply x-pack/plugins/actions/server/builtin_action_types/torq.test.ts:146]: Non-emptiness is validated in the UI, here it doesn't matter if the body is empty.

File: x-pack/plugins/actions/server/builtin_action_types/torq.ts (hunk 22 above)
Comment (natexcvi) [reply x-pack/plugins/actions/server/builtin_action_types/torq.ts:176]: It is

File: x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx (hunk 25 above)
Comment (natexcvi) [reply x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/torq/torq.tsx]: Exactly like this, followed their example

File: x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx (hunk 34 above)
Comment (natexcvi) [reply x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx:363]: There were inconsistencies in the signature of that function

File: x-pack/plugins/triggers_actions_ui/public/application/lib/get_defaults_for_action_params.ts (hunk 33 above)
Comment (natexcvi) [reply x-pack/plugins/triggers_actions_ui/public/application/lib/get_defaults_for_action_params.ts]: Don't you already know we're special?
